src/IPStack/Packet/IPPacket.swift

```diff
@@ -271,17 +271,19 @@ open class IPPacket {
         setPayloadWithUInt16(offset, at: 6)
         setPayloadWithUInt8(TTL, at: 8)
         setPayloadWithUInt8(transportProtocol.rawValue, at: 9)
-        // clear checksum bytes
-        resetPayloadAt(10, length: 2)
+
         setPayloadWithUInt32(sourceAddress.UInt32InNetworkOrder!, at: 12, swap: false)
         setPayloadWithUInt32(destinationAddress.UInt32InNetworkOrder!, at: 16, swap: false)
 
+        // set IP checksum
+        resetPayloadAt(10, length: 2)
+        setPayloadWithUInt16(Checksum.computeChecksum(packetData, from: 0, to: Int(headerLength)), at: 10, swap: false)
+        
```

Comment:
Yes, you are right. I review the code, the problem is here:

```
        var checksum = Checksum.computeChecksum(packetData, from: 0, to: nil, withPseudoHeaderChecksum: pseudoHeaderChecksum)
```

from need from UDP Header

```
        var checksum = Checksum.computeChecksum(packetData, from: offset, to: nil, withPseudoHeaderChecksum: pseudoHeaderChecksum)
```

This is a coincidence

When I make the change, I change the checksum order make it work, and didn't think more, sorry.